test/parse/test_entity_declaration.rb

```diff
@@ -23,6 +23,159 @@ def parse(internal_subset)
     end
 
     public
+
+    class TestGeneralEntityDeclration < self
+      def test_invalid_entity_definition
+        exception = assert_raise(REXML::ParseException) do
+          REXML::Document.new('<!DOCTYPE root [<!ENTITY "valid-name" invalid-entity-definition > ]>')
+        end
+        assert_equal(<<-DETAIL.chomp, exception.to_s)
+Malformed entity declaration
+Line: 1
+Position: 68
+Last 80 unconsumed characters:
+ \"valid-name\" invalid-entity-definition > ]>
+        DETAIL
+      end
+
+      class TestSystemExternalEntities < self
+        def test_invalid_external_entity_before_system_literal
+          exception = assert_raise(REXML::ParseException) do
+            REXML::Document.new('<!DOCTYPE root [<!ENTITY valid-name SYSTEM invalid-external-entity "valid-system-literal" > ]>')
+          end
+          assert_equal(<<-DETAIL.chomp, exception.to_s)
+Malformed entity declaration
+Line: 1
+Position: 94
+Last 80 unconsumed characters:
+ valid-name SYSTEM invalid-external-entity \"valid-system-literal\" > ]>
+          DETAIL
+        end
+
+        def test_invalid_external_entity_after_system_literal
```

Comment:
Could you split this to `TestNotationDataDeclaration`?

```suggestion
        def test_not_ndata
```